test_runner/src/main/kotlin/ftl/args/ValidateCommonArgs.kt

```diff
@@ -0,0 +1,55 @@
+package ftl.args
+
+import ftl.config.FtlConstants
+import ftl.util.FlankFatalError
+
+fun CommonArgs.validate() {
+    assertProjectId()
+    assertMaxTestShards()
+    assertShardTime()
+    assertRepeatTests()
+    assertSmartFlankGcsPath()
+}
+
+private fun CommonArgs.assertProjectId() {
+    if (project.isEmpty()) throw FlankFatalError(
+        "The project is not set. Define GOOGLE_CLOUD_PROJECT, set project in flank.yml\n" +
+                "or save service account credential to ${FtlConstants.defaultCredentialPath}\n" +
+                " See https://github.com/GoogleCloudPlatform/google-cloud-java#specifying-a-project-id"
+    )
+}
+
+private fun CommonArgs.assertMaxTestShards() {
+    if (
+        maxTestShards !in IArgs.AVAILABLE_SHARD_COUNT_RANGE &&
+        maxTestShards != -1
+    ) throw FlankFatalError(
+        "max-test-shards must be >= 1 and <= 50, or -1. But current is $maxTestShards"
```

Comment:
I think we should use IArgs.AVAILABLE_SHARD_COUNT_RANGE.first / IArgs.AVAILABLE_SHARD_COUNT_RANGE.last in exception message.